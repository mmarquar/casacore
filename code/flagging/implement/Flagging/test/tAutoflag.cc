#include <casa/namespace.h>
int main (int,char *)
{
  return 0;
}
