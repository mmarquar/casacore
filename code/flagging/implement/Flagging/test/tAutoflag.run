#!/bin/sh
#-----------------------------------------------------------------------------
# Usage: tAutoflag.run
#-----------------------------------------------------------------------------
# This script executes the tAutoflag.g test script to check if all the
# flagging modules are working
#
# $Id$
#-----------------------------------------------------------------------------
  if [ "$AIPSPATH" = "" ]
  then
     echo "UNTESTED: tAutoflag.run (AIPSPATH not defined)"
     exit 3
  fi

  # establish location of test MS
  MS='WSRT-test.MS2'
  AIPSDEMO=`echo $AIPSPATH | awk '{printf("%s/data/demo/autoflag",$1)}'`
  MSPATH="$AIPSDEMO/$MS"
  if [ ! -d "$MSPATH" ]; then
    MSPATH="$HOME/aips++/data/demo/autoflag/$MS"
    if [ ! -d "$MSPATH" ]; then
      echo "UNTESTED: tAutoflag.run (found neither $AIPSDEMO/$MS nor $MSPATH)"
      exit 3
    fi
  fi
  # set symlink
  LINK='tAutoflag.MS2'
  rm -f $LINK 2>/dev/null
  if ! ln -s $MSPATH $LINK; then
    echo "UNTESTED: tAutoflag.run (failed to set $LINK symlink)"
    exit 3
  fi
  
  # run the glish script
  echo ">>>"
  glish -l tAutoflag.g
