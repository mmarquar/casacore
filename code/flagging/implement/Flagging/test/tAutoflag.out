>>>

 glishrc: mode= Operational (Dwingeloo)

 Path used by Glish (system.path.include, set in .glishrc):
.
/aips++/jnoordam/wbork/
/aips++/daily/dop07_egcs/libexec
system.path.glishelp: /aips++/jnoordam/wbork/
 
NORMAL: dowait variable is T
NORMAL: Pre-starting timer misc quanta
NORMAL: Starting server timer
NORMAL: Starting server misc
NORMAL: Starting server quanta
NORMAL: defaultservers ready
NORMAL: Starting server autoflag
NORMAL: Server started: /dop07_0/oms/aips++/code/trial/implement/Flagging/test/autoflag (AIPS++ version: 1.6 (build #37))
NORMAL: Server started: quanta (AIPS++ version: 1.6 (build #42))
NORMAL: Server started: misc (AIPS++ version: 1.6 (build #42))
NORMAL: attached MS tAutoflag.MS2: 6656 rows, 104 times, 105 IFRs
<<<
Added method 1: timemed (TimeMedian)
    thr         = 5
    hw          = 10
    rowthr      = 10
    rowhw       = 6
    column      = DATA
    expr        = + ABS XX YY
    debug       = F
    fignore     = F
Added method 2: freqmed (FreqMedian)
    thr         = 5
    hw          = 10
    rowthr      = 10
    rowhw       = 6
    column      = DATA
    expr        = + ABS XX YY
    debug       = F
    fignore     = F
Added method 3: sprej (SpectralRej)
    region      = F
    spwid       = F
    fq          = F
   *chan        = [5 30] 
    ndeg        = 2
    rowthr      = 5
    rowhw       = 6
    column      = DATA
    expr        = + ABS XX YY
    debug       = F
    fignore     = F
Added method 4: uvbin (UVBinner)
   *thr         = 0.04
    nbins       = 50
   *plotchan    = 100
    econoplot   = T
    column      = DATA
    expr        = + ABS XX YY
    fignore     = F
Added method 5: select (Selector)
    spwid       = F
    field       = F
   *fq          = [806 808] 
    chan        = F
    corr        = F
   *ant         = RT6
    baseline    = F
    timerng     = F
    autocorr    = F
    timeslot    = F
    dtime       = 10
    quack       = F
    clip        = F
    flagrange   = F
    unflag      = F
Added method 6: select (Selector)
    spwid       = F
    field       = F
    fq          = F
    chan        = F
    corr        = F
    ant         = F
    baseline    = F
    timerng     = F
    autocorr    = F
    timeslot    = F
    dtime       = 10
    quack       = F
   *clip        = [expr=+ ABS XX YY, max=10]
    flagrange   = F
    unflag      = F
NORMAL: Starting flagger::run
NORMAL: Existing flags will be reset
NORMAL: (DATA)ABS(XX)+ABS(YY); thr=5.0 rowthr=10.0  hw=10
NORMAL: Existing flags will be reset
NORMAL: (DATA)ABS(XX)+ABS(YY); thr=5.0 rowthr=10.0  hw=10
NORMAL: Existing flags will be reset
NORMAL: (DATA)ABS(XX)+ABS(YY); #4-29; ndeg=2 rowthr=5.0 rowhw=6 
NORMAL: Existing flags will be reset
NORMAL: (DATA)ABS(XX)+ABS(YY); thr=0.04 nbins=50,50 
NORMAL: Existing flags will be reset
NORMAL: fq=806.00-808.00MHz ant=RT6; flag all 
NORMAL: Existing flags will be reset
NORMAL: flag ABS(XX)+ABS(YY)>10 
NORMAL: Chunk 1: correlations XX YY, 256 chans, 104 times, 6656 rows
NORMAL: 26 channels will be fitted in this chunk
NORMAL: defaultaipsrc (aipsrc()) ready
NORMAL: defaultwidgetserver (dws) ready
NORMAL: TimeMedian: 15649 pixel flags, 1455 row flags
NORMAL: FreqMedian: 24443 pixel flags, 947 row flags
NORMAL: SpectralRej: 0 pixel flags, 4893 row flags
NORMAL: UVBinner: 63652 pixel flags, 0 row flags
NORMAL: Selector: 53040 pixel flags, 0 row flags
NORMAL: Selector#2: 46764 pixel flags, 0 row flags
NORMAL: Flagging complete
NORMAL: 
>>>
NORMAL: Finished flagger::run
      80.14 real       61.98 user        0.81 system
<<<
